File: aoc/aoc25/d1/part1.py
import sys

p = 50
N = 100
s = 0
for line in sys.stdin:
    dir = -1 if line[0] == "L" else 1
    x = int(line[1:])
    p = (p + dir*x) % N
    if p == 0:
        s += 1

print(s)

File: aoc/aoc24/d22/part1.rs
const MOD: u64 = 16777216;
const STEPS: u32 = 2000;

fn evolve(mut bit_string: u64) -> u64 {
    for _ in 0..STEPS {
        bit_string ^= bit_string * 64;
        bit_string %= MOD;
        
        bit_string ^= bit_string / 32;
        bit_string %= MOD;
        
        bit_string ^= bit_string * 2048;
        bit_string %= MOD;
    }
    bit_string
}

fn main() {
    let mut total = 0;
    for line in std::io::stdin().lines().map(|x| x.unwrap()) {
        let bit_string = line.parse().unwrap();
        total += evolve(bit_string);
    }
    println!("{}", total)

}